mcs/tools/al/Al.cs

```diff
@@ -765,6 +729,83 @@ private void SetKeyPair (AssemblyName aname)
 			}
 		}
 
+		private void ReadCustomAttributesFromTemplateFile (string templateFile, ref AssemblyName aname)
+		{
+			// LAMESPEC: according to MSDN, the template assembly must have a
+			// strong name but this is not enforced
+			var options = IKR.UniverseOptions.MetadataOnly;
+
+			var universe = new IKR.Universe (options);
+			var asm = universe.LoadFile (templateFile);
+
+			// Create missing assemblies, we don't want to load them!
+			// Code taken from ikdasm
+			var names = new HashSet<string> ();
+			IKR.AssemblyName[] assembly_refs = asm.ManifestModule.__GetReferencedAssemblies ();
+
+			var resolved_assemblies = new IKR.Assembly [assembly_refs.Length];
+			for (int i = 0; i < resolved_assemblies.Length; i++) {
+				string name = assembly_refs [i].Name;
+
+				while (names.Contains (name)) {
+					name = name + "_" + i;
+				}
+				names.Add (name);
+				resolved_assemblies [i] = universe.CreateMissingAssembly (assembly_refs [i].FullName);
+			}
+			asm.ManifestModule.__ResolveReferencedAssemblies (resolved_assemblies);
+
+			foreach (var attr_data in asm.__GetCustomAttributes (null, false)) {
+				string asm_name = attr_data.AttributeType.Assembly.GetName ().Name;
```

Comment:
Using `Universe::Mscorlib` would not be correct here, because the template assembly might not be referencing the same mscorlib as the one used by the executing assembly. Eg. if the template assembly is from `Xamarin.Android`, then the versions wouldn't match. In any case, we don't need to match the version. We just want to check that the attribute is from an assembly named `mscorlib`.